ladybug_geometry/geometry2d/mesh.py

```diff
@@ -9,6 +9,13 @@
 from .polygon import Polygon2D
 from ._2d import Base2DIn2D
 
+try:
```

Comment:
Yes, we can move this under the `from_dict` method and throw an exception only in the case that `'colors'` are in the input data dictionary. 

And, yes, the main reason why Color is not available by default is that it that would create a circular dependency between `ladybug` core and `ladybug_geometry`.  However, it's also because this is essentially the only outside dependency of the entire `ladybug_geometry` library and it is for a completely optional property of the mesh.  Particularly because colors are more for visual presentation that's added at the end of studies, I imagine that most meshes won't have colors and so it doesn't seem to make sense to add an entire dependency for something that only affects a few special cases.